VAMobile/src/screens/HomeScreen/ProfileScreen/ContactInformationScreen/EditPhoneNumberScreen/EditPhoneNumberScreen.test.tsx

```diff
@@ -31,180 +27,75 @@ context('EditPhoneNumberScreen', () => {
       {
         params: {
           displayTitle: 'Home phone',
-          phoneType: 'HOME',
           phoneData,
+          ...data
         },
       },
     )
+    
 
-    store = {
-      ...InitialState,
-      errors: errorsState,
-    }
-
-    component = render(<EditPhoneNumberScreen {...props} />, { preloadedState: store })
-
-    testInstance = component.UNSAFE_root
+    render(<EditPhoneNumberScreen {...props} />)
   }
 
   beforeEach(() => {
-    initializeTestInstance({
-      id: 0,
-      areaCode: '858',
-      phoneNumber: '1234567',
-      countryCode: '1',
-      phoneType: 'HOME',
-    })
-  })
-
-  it('initializes correctly', async () => {
-    expect(component).toBeTruthy()
+    renderWithData()
   })
 
   describe('when the phone number exists', () => {
-    it('should display the remove button', () => {
-      initializeTestInstance({
-        id: 0,
-        areaCode: '858',
-        phoneNumber: '1234567',
-        countryCode: '1',
-        phoneType: 'HOME',
-      })
-      const buttons = testInstance.findAllByType(VAButton)
-      expect(buttons[buttons.length - 1].props.label).toEqual('Remove home phone')
+    it('displays the remove button', () => {
+      renderWithData()
+      expect(screen.getByRole('button', { name: 'Remove home phone' })).toBeTruthy()
     })
   })
 
-  describe('when the text input changes', () => {
+  describe('Phone number text input', () => {
     describe('when the length is less than or equal to 10 digits', () => {
-      it('should display just the numbers in the text input', async () => {
-        const phoneNumTextInput = testInstance.findAllByType(TextInput)[0]
-        phoneNumTextInput.props.onChangeText('12345')
-        expect(phoneNumTextInput.props.value).toEqual('12345')
+      it('displays just the numbers in the text input', async () => {
+        const phoneNumberInput = screen.getByTestId('phoneNumberTestID')
+
+        fireEvent.changeText(phoneNumberInput, '12345')
+        phoneNumberInput.props.onEndEditing()
```

Comment:
I feel the same way. When I was working on this, I initially tried to interact with just the UI, but the formatting changes to the number wouldn't happen unless the `onEndEditing` prop was fired off 🙁. I tried tapping into other elements after changing the phone number input, but still no luck. In `EditPhoneNumberScreen`, the formatting happens once `onEndEditing` is called, so it seems that might not be happening with `fireEvent` for some reason. I wonder if it would work with [user-event](https://github.com/testing-library/user-event) library, which is what RTL recommends for simulating user events